File: id-card-toolkit-ios-sdk-v3.1.6/id-card-toolkit-ios-sdk-v3.1.6/samples/IDCardToolkitObjCApp/IDCardToolkitObjCApp/ConversionClass/PublicDataParse.m
//
//  PublicDataParse.m
//  EIDAToolkitObjCTestapp
//
//  Created by Federal Authority For Identity and Citizenship on 07/11/17.
//  Copyright © 2017 Federal Authority For Identity and Citizenship. All rights reserved.
//

#import "PublicDataParse.h"

@implementation PublicDataParse

-(id)init {
    self=[super init];
    if (self) {
        
        self.pModDataDictionary=[[NSMutableDictionary alloc]init];
        self.pNonModDataDictionary=[[NSMutableDictionary alloc]init];
        self.pHomeAddressDictionary=[[NSMutableDictionary alloc]init];
        self.pWorkAddressDictionary=[[NSMutableDictionary alloc]init];
    }
    return self;
}
-(NSMutableDictionary *)getModifiablePublicDataDetails:(ModifiablePublicData *)modData {
    
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getOccupationCode]] forKey:@"OccupationCode"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getOccupationArabic]] forKey:@"OccupationArabic"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getOccupationEnglish]] forKey:@"OccupationEnglish"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getFamilyId]] forKey:@"FamilyId"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getOccupationTypeArabic]] forKey:@"OccupationTypeArabic"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getOccupationTypeEnglish]] forKey:@"OccupationTypeEnglish"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getOccupationFieldCode]] forKey:@"OccupationFieldCode"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getCompanyNameArabic]] forKey:@"CompanyNameArabic"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getCompanyNameEnglish]] forKey:@"CompanyNameEnglish"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getMaritalStatusCode]] forKey:@"MaritalStatusCode"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getHusbandIdnNumber]] forKey:@"HusbandIdNumber"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getSponsorTypeCode]] forKey:@"SponsorTypeCode"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getSponsorUnifiedNumber]] forKey:@"SponsorUnifiedNumber"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getSponsorName]] forKey:@"SponsorName"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getResidencyTypeCode]] forKey:@"ResidencyTypeCode"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getResidencyNumber]] forKey:@"ResidencyNumber"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getResidencyExpiryDate]] forKey:@"ResidencyExpiryDate"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getPassportNumber]] forKey:@"PassportNumber"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getPassportTypeCode]] forKey:@"PassportTypeCode"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getPassportCountryCode]] forKey:@"PassportCountryCode"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getPassportCountryArabic]] forKey:@"PassportCountryArabic"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getPassportCountryEnglish]] forKey:@"PassportCountryEnglish"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getPassportIssueDate]] forKey:@"PassportIssueDate"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getPassportExpiryDate]] forKey:@"PassportExpiryDate"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getQualificationLevelCode]] forKey:@"QualificationLevelCode"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getQualificationLevelArabic]] forKey:@"QualificationLevelArabic"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getQualificationLevelEnglish]] forKey:@"QualificationLevelEnglish"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getDegreeDescArabic]] forKey:@"DegreeDescArabic"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getDegreeDescEnglish]] forKey:@"DegreeDescEnglish"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getFieldOfStudyCode]] forKey:@"FieldOfStudyCode"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getFieldOfStudyArabic]] forKey:@"FieldOfStudyArabic"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getFieldOfStudyEnglish]] forKey:@"FieldOfStudyEnglish"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getPlaceOfStudyArabic]] forKey:@"PlaceOfStudyArabic"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getPlaceOfStudyEnglish]] forKey:@"PlaceOfStudyEnglish"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getDateOfGraduation]] forKey:@"DateOfGraduation"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getMotherFullNameArabic]] forKey:@"MotherFullNameArabic"];
    [self.pModDataDictionary setValue:[NSString stringWithFormat:@"%@",[modData getMotherFullNameEnglish]] forKey:@"MotherFullNameEnglish"];
    
    return  self.pModDataDictionary;
}
-(NSMutableDictionary *)getNonModifiablePublicDataDetails:(NonModifiablePublicData *)nonModData {
    
    [self.pNonModDataDictionary setValue:[NSString stringWithFormat:@"%@",[nonModData getIDType]] forKey:@"IDType"];
    [self.pNonModDataDictionary setValue:[NSString stringWithFormat:@"%@",[nonModData getIssueDate]] forKey:@"IssueDate"];
    [self.pNonModDataDictionary setValue:[NSString stringWithFormat:@"%@",[nonModData getExpiryDate]] forKey:@"ExpiryDate"];
    [self.pNonModDataDictionary setValue:[NSString stringWithFormat:@"%@",[nonModData getTitleArabic]] forKey:@"TitleArabic"];
    [self.pNonModDataDictionary setValue:[NSString stringWithFormat:@"%@",[nonModData getFullNameArabic]] forKey:@"FullNameArabic"];
    [self.pNonModDataDictionary setValue:[NSString stringWithFormat:@"%@",[nonModData getTitleEnglish]] forKey:@"TitleEnglish"];
    [self.pNonModDataDictionary setValue:[NSString stringWithFormat:@"%@",[nonModData getFullNameEnglish]] forKey:@"FullNameEnglish"];
    [self.pNonModDataDictionary setValue:[NSString stringWithFormat:@"%@",[nonModData getGender]] forKey:@"Gender"];
    [self.pNonModDataDictionary setValue:[NSString stringWithFormat:@"%@",[nonModData getNationalityArabic]] forKey:@"NationalityArabic"];
    [self.pNonModDataDictionary setValue:[NSString stringWithFormat:@"%@",[nonModData getNationalityEnglish]] forKey:@"NationalityEnglish"];
    [self.pNonModDataDictionary setValue:[NSString stringWithFormat:@"%@",[nonModData getNationalityCode]] forKey:@"NationalityCode"];
    [self.pNonModDataDictionary setValue:[NSString stringWithFormat:@"%@",[nonModData getDateOfBirth]] forKey:@"DateOfBirth"];
    [self.pNonModDataDictionary setValue:[NSString stringWithFormat:@"%@",[nonModData getPlaceOfBirthArabic]] forKey:@"PlaceOfBirthArabic"];
    [self.pNonModDataDictionary setValue:[NSString stringWithFormat:@"%@",[nonModData getPlaceOfBirthEnglish]] forKey:@"PlaceOfBirthEnglish"];
    
    return  self.pNonModDataDictionary;
}
-(NSMutableDictionary *)getHomeAddressDetails:(HomeAddress *)homeaddress {
    
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getAddressTypeCode]] forKey:@"AddressTypeCode"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getLocationCode]] forKey:@"LocationCode"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getEmiratesCode]] forKey:@"EmiratesCode"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getEmiratesArabic]] forKey:@"EmiratesArabic"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getEmiratesEnglish]] forKey:@"EmiratesEnglish"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getCityCode]] forKey:@"CityCode"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getCityArabic]] forKey:@"CityArabic"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getCityEnglish]] forKey:@"CityEnglish"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getStreetArabic]] forKey:@"StreetArabic"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getStreetEnglish]] forKey:@"StreetEnglish"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getPobox]] forKey:@"Pobox"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getAreaCode]] forKey:@"AreaCode"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getAreaArabic]] forKey:@"AreaArabic"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getAreaEnglish]] forKey:@"AreaEnglish"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getBuildingNameArabic]] forKey:@"BuildingNameArabic"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getBuildingNameEnglish]] forKey:@"BuildingNameEnglish"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getLandPhoneNumber]] forKey:@"ResidentPhoneNumber"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getMobilePhoneNumber]] forKey:@"MobilePhoneNumber"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getEmail]] forKey:@"Email"];
    [self.pHomeAddressDictionary setValue:[NSString stringWithFormat:@"%@",[homeaddress getFlatNumber]] forKey:@"FlatNumber"];
    
    return self.pHomeAddressDictionary;
}
-(NSMutableDictionary *)getWorkAddressDetails:(WorkAddress *)workAddress {
    
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getAddressTypeCode]] forKey:@"AddressTypeCode"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getLocationCode]] forKey:@"LocationCode"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getCompanyNameArabic]] forKey:@"CompanyNameArabic"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getCompanyNameEnglish]] forKey:@"CompanyNameEnglish"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getEmiratesCode]] forKey:@"EmiratesCode"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getEmiratesArabic]] forKey:@"EmiratesArabic"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getEmiratesEnglish]] forKey:@"EmiratesEnglish"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getCityCode]] forKey:@"CityCode"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getCityArabic]] forKey:@"CityArabic"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getCityEnglish]] forKey:@"CityEnglish"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getPoBox]] forKey:@"PoBox"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getStreetArabic]] forKey:@"StreetArabic"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getStreetEnglish]] forKey:@"StreetEnglish"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getAreaCode]] forKey:@"AreaCode"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getAreaArabic]] forKey:@"AreaArabic"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getAreaEnglish]] forKey:@"AreaEnglish"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getBuildingNameArabic]] forKey:@"BuildingNameArabic"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getBuildingNameEnglish]] forKey:@"BuildingNameEnglish"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getLandPhoneNumber]] forKey:@"LandPhoneNumber"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getMobilePhoneNumber]] forKey:@"MobilePhoneNumber"];
    [self.pWorkAddressDictionary setValue:[NSString stringWithFormat:@"%@",[workAddress getEmail]] forKey:@"Email"];
    
    return self.pWorkAddressDictionary;
}
@end

File: id-card-toolkit-ios-sdk-v3.1.6/id-card-toolkit-ios-sdk-v3.1.6/samples/ICASwiftApp/ICASwiftApp/ConversionClass/PublicDataParse.swift
//
//  PublicDataParse.swift
//  IDCardToolkitSwiftApp
//
//  Created by Federal Authority For Identity and Citizenship on 23/12/17.
//  Copyright © 2017 Federal Authority For Identity and Citizenship. All rights reserved.
//

import UIKit

class PublicDataParse: NSObject {

    var pModDataDictionary = NSMutableDictionary()
    var pNonModDataDictionary = NSMutableDictionary()
    var pHomeAddressDictionary = NSMutableDictionary()
    var pWorkAddressDictionary = NSMutableDictionary()
    
    func getModifiablePublicDataDetails(_ modData:ModifiablePublicData) -> NSMutableDictionary {
        
        var resultData:String
        resultData = modData.getOccupationCode()
        self.pModDataDictionary["OccupationCode"] = "\(resultData)"
        resultData = modData.getOccupationArabic()
        self.pModDataDictionary["OccupationArabic"] = "\(resultData)"
        resultData = modData.getOccupationEnglish()
        self.pModDataDictionary["OccupationEnglish"] = "\(resultData)"
        resultData = modData.getFamilyId()
        self.pModDataDictionary["FamilyId"] = "\(resultData)"
        resultData = modData.getOccupationTypeArabic()
        self.pModDataDictionary["OccupationTypeArabic"] = "\(resultData)"
        resultData = modData.getOccupationTypeEnglish()
        self.pModDataDictionary["OccupationTypeEnglish"] = "\(resultData)"
        resultData = modData.getOccupationFieldCode()
        self.pModDataDictionary["OccupationFieldCode"] = "\(resultData)"
        resultData = modData.getCompanyNameArabic()
        self.pModDataDictionary["CompanyNameArabic"] = "\(resultData)"
        resultData = modData.getCompanyNameEnglish()
        self.pModDataDictionary["CompanyNameEnglish"] = "\(resultData)"
        resultData = modData.getMaritalStatusCode()
        self.pModDataDictionary["MaritalStatusCode"] = "\(resultData)"
        resultData = modData.getHusbandIdnNumber()
        self.pModDataDictionary["HusbandIdNumber"] = "\(resultData)"
        resultData = modData.getSponsorTypeCode()
        self.pModDataDictionary["SponsorTypeCode"] = "\(resultData)"
        resultData = modData.getSponsorUnifiedNumber()
        self.pModDataDictionary["SponsorUnifiedNumber"] = "\(resultData)"
        resultData = modData.getSponsorName()
        self.pModDataDictionary["SponsorName"] = "\(resultData)"
        resultData = modData.getResidencyTypeCode()
        self.pModDataDictionary["ResidencyTypeCode"] = "\(resultData)"
        resultData = modData.getResidencyNumber()
        self.pModDataDictionary["ResidencyNumber"] = "\(resultData)"
        resultData = modData.getResidencyExpiryDate()
        self.pModDataDictionary["ResidencyExpiryDate"] = "\(resultData)"
        resultData = modData.getPassportNumber()
        self.pModDataDictionary["PassportNumber"] = "\(resultData)"
        resultData = modData.getPassportTypeCode()
        self.pModDataDictionary["PassportTypeCode"] = "\(resultData)"
        resultData = modData.getPassportCountryCode()
        self.pModDataDictionary["PassportCountryCode"] = "\(resultData)"
        resultData = modData.getPassportCountryArabic()
        self.pModDataDictionary["PassportCountryArabic"] = "\(resultData)"
        resultData = modData.getPassportCountryEnglish()
        self.pModDataDictionary["PassportCountryEnglish"] = "\(resultData)"
        resultData = modData.getPassportIssueDate()
        self.pModDataDictionary["PassportIssueDate"] = "\(resultData)"
        resultData = modData.getPassportExpiryDate()
        self.pModDataDictionary["PassportExpiryDate"] = "\(resultData)"
        resultData = modData.getQualificationLevelCode()
        self.pModDataDictionary["QualificationLevelCode"] = "\(resultData)"
        resultData = modData.getQualificationLevelArabic()
        self.pModDataDictionary["QualificationLevelArabic"] = "\(resultData)"
        resultData = modData.getQualificationLevelEnglish()
        self.pModDataDictionary["QualificationLevelEnglish"] = "\(resultData)"
        resultData = modData.getDegreeDescArabic()
        self.pModDataDictionary["DegreeDescArabic"] = "\(resultData)"
        resultData = modData.getDegreeDescEnglish()
        self.pModDataDictionary["DegreeDescEnglish"] = "\(resultData)"
        resultData = modData.getFieldOfStudyCode()
        self.pModDataDictionary["FieldOfStudyCode"] = "\(resultData)"
        resultData = modData.getFieldOfStudyArabic()
        self.pModDataDictionary["FieldOfStudyArabic"] = "\(resultData)"
        resultData = modData.getFieldOfStudyEnglish()
        self.pModDataDictionary["FieldOfStudyEnglish"] = "\(resultData)"
        resultData = modData.getPlaceOfStudyArabic()
        self.pModDataDictionary["PlaceOfStudyArabic"] = "\(resultData)"
        resultData = modData.getPlaceOfStudyEnglish()
        self.pModDataDictionary["PlaceOfStudyEnglish"] = "\(resultData)"
        resultData = modData.getDateOfGraduation()
        self.pModDataDictionary["DateOfGraduation"] = "\(resultData)"
        resultData = modData.getMotherFullNameArabic()
        self.pModDataDictionary["MotherFullNameArabic"] = "\(resultData)"
        resultData = modData.getMotherFullNameEnglish()
        self.pModDataDictionary["MotherFullNameEnglish"] = "\(resultData)"
        
        return self.pModDataDictionary
    }
    func getNonModifiablePublicDataDetails(_ nonModData:NonModifiablePublicData) -> NSMutableDictionary {

        var resultData:String
        resultData = nonModData.getIDType()
        self.pNonModDataDictionary["IDType"] = "\(resultData)"
        resultData = nonModData.getIssueDate()
        self.pNonModDataDictionary["IssueDate"] = "\(resultData)"
        resultData = nonModData.getExpiryDate()
        self.pNonModDataDictionary["ExpiryDate"] = "\(resultData)"
        resultData = nonModData.getTitleArabic()
        self.pNonModDataDictionary["TitleArabic"] = "\(resultData)"
        resultData = nonModData.getFullNameArabic()
        self.pNonModDataDictionary["FullNameArabic"] = "\(resultData)"
        resultData = nonModData.getTitleEnglish()
        self.pNonModDataDictionary["TitleEnglish"] = "\(resultData)"
        resultData = nonModData.getFullNameEnglish()
        self.pNonModDataDictionary["FullNameEnglish"] = "\(resultData)"
        resultData = nonModData.getGender()
        self.pNonModDataDictionary["Gender"] = "\(resultData)"
        resultData = nonModData.getNationalityArabic()
        self.pNonModDataDictionary["NationalityArabic"] = "\(resultData)"
        resultData = nonModData.getNationalityEnglish()
        self.pNonModDataDictionary["NationalityEnglish"] = "\(resultData)"
        resultData = nonModData.getNationalityCode()
        self.pNonModDataDictionary["NationalityCode"] = "\(resultData)"
        resultData = nonModData.getDateOfBirth()
        self.pNonModDataDictionary["DateOfBirth"] = "\(resultData)"
        resultData = nonModData.getPlaceOfBirthArabic()
        self.pNonModDataDictionary["PlaceOfBirthArabic"] = "\(resultData)"
        resultData = nonModData.getPlaceOfBirthEnglish()
        self.pNonModDataDictionary["PlaceOfBirthEnglish"] = "\(resultData)"

        return self.pNonModDataDictionary
    }
    func getHomeAddressDetails(_ homeaddress:HomeAddress) -> NSMutableDictionary {
        
        var resultData:String
        resultData = homeaddress.getAddressTypeCode()
        self.pHomeAddressDictionary["AddressTypeCode"] = "\(resultData)"
        resultData = homeaddress.getLocationCode()
        self.pHomeAddressDictionary["LocationCode"] = "\(resultData)"
        resultData = homeaddress.getEmiratesCode()
        self.pHomeAddressDictionary["EmiratesCode"] = "\(resultData)"
        resultData = homeaddress.getEmiratesArabic()
        self.pHomeAddressDictionary["EmiratesArabic"] = "\(resultData)"
        resultData = homeaddress.getEmiratesEnglish()
        self.pHomeAddressDictionary["EmiratesEnglish"] = "\(resultData)"
        resultData = homeaddress.getCityCode()
        self.pHomeAddressDictionary["CityCode"] = "\(resultData)"
        resultData = homeaddress.getCityArabic()
        self.pHomeAddressDictionary["CityArabic"] = "\(resultData)"
        resultData = homeaddress.getCityEnglish()
        self.pHomeAddressDictionary["CityEnglish"] = "\(resultData)"
        resultData = homeaddress.getStreetArabic()
        self.pHomeAddressDictionary["StreetArabic"] = "\(resultData)"
        resultData = homeaddress.getStreetEnglish()
        self.pHomeAddressDictionary["StreetEnglish"] = "\(resultData)"
        resultData = homeaddress.getPobox()
        self.pHomeAddressDictionary["Pobox"] = "\(resultData)"
        resultData = homeaddress.getAreaCode()
        self.pHomeAddressDictionary["AreaCode"] = "\(resultData)"
        resultData = homeaddress.getAreaArabic()
        self.pHomeAddressDictionary["AreaArabic"] = "\(resultData)"
        resultData = homeaddress.getAreaEnglish()
        self.pHomeAddressDictionary["AreaEnglish"] = "\(resultData)"
        resultData = homeaddress.getBuildingNameArabic()
        self.pHomeAddressDictionary["BuildingNameArabic"] = "\(resultData)"
        resultData = homeaddress.getBuildingNameEnglish()
        self.pHomeAddressDictionary["BuildingNameEnglish"] = "\(resultData)"
        resultData = homeaddress.getLandPhoneNumber()
        self.pHomeAddressDictionary["ResidentPhoneNumber"] = "\(resultData)"
        resultData = homeaddress.getMobilePhoneNumber()
        self.pHomeAddressDictionary["MobilePhoneNumber"] = "\(resultData)"
        resultData = homeaddress.getEmail()
        self.pHomeAddressDictionary["Email"] = "\(resultData)"
        resultData = homeaddress.getFlatNumber()
        self.pHomeAddressDictionary["FlatNumber"] = "\(resultData)"
        
         return self.pHomeAddressDictionary

    }
    func getWorkAddressDetails(_ workAddress:WorkAddress) -> NSMutableDictionary {

        var resultData:String
        resultData = workAddress.getAddressTypeCode()
        self.pWorkAddressDictionary["AddressTypeCode"] = "\(resultData)"
        resultData = workAddress.getLocationCode()
        self.pWorkAddressDictionary["LocationCode"] = "\(resultData)"
        resultData = workAddress.getCompanyNameArabic()
        self.pWorkAddressDictionary["CompanyNameArabic"] = "\(resultData)"
        resultData = workAddress.getCompanyNameEnglish()
        self.pWorkAddressDictionary["CompanyNameEnglish"] = "\(resultData)"
        resultData = workAddress.getEmiratesCode()
        self.pWorkAddressDictionary["EmiratesCode"] = "\(resultData)"
        resultData = workAddress.getEmiratesArabic()
        self.pWorkAddressDictionary["EmiratesArabic"] = "\(resultData)"
        resultData = workAddress.getEmiratesEnglish()
        self.pWorkAddressDictionary["EmiratesEnglish"] = "\(resultData)"
        resultData = workAddress.getCityCode()
        self.pWorkAddressDictionary["CityCode"] = "\(resultData)"
        resultData = workAddress.getCityArabic()
        self.pWorkAddressDictionary["CityArabic"] = "\(resultData)"
        resultData = workAddress.getCityEnglish()
        self.pWorkAddressDictionary["CityEnglish"] = "\(resultData)"
        resultData = workAddress.getPoBox()
        self.pWorkAddressDictionary["PoBox"] = "\(resultData)"
        resultData = workAddress.getStreetArabic()
        self.pWorkAddressDictionary["StreetArabic"] = "\(resultData)"
        resultData = workAddress.getStreetEnglish()
        self.pWorkAddressDictionary["StreetEnglish"] = "\(resultData)"
        resultData = workAddress.getAreaCode()
        self.pWorkAddressDictionary["AreaCode"] = "\(resultData)"
        resultData = workAddress.getAreaArabic()
        self.pWorkAddressDictionary["AreaArabic"] = "\(resultData)"
        resultData = workAddress.getAreaEnglish()
        self.pWorkAddressDictionary["AreaEnglish"] = "\(resultData)"
        resultData = workAddress.getBuildingNameArabic()
        self.pWorkAddressDictionary["BuildingNameArabic"] = "\(resultData)"
        resultData = workAddress.getBuildingNameEnglish()
        self.pWorkAddressDictionary["BuildingNameEnglish"] = "\(resultData)"
        resultData = workAddress.getLandPhoneNumber()
        self.pWorkAddressDictionary["LandPhoneNumber"] = "\(resultData)"
        resultData = workAddress.getMobilePhoneNumber()
        self.pWorkAddressDictionary["MobilePhoneNumber"] = "\(resultData)"
        resultData = workAddress.getEmail()
        self.pWorkAddressDictionary["Email"] = "\(resultData)"
    
        return self.pWorkAddressDictionary
   }


}






